index.js

```diff
@@ -134,83 +612,58 @@ app.intent('send.message - yes', async conv => {
 app.intent('send.message - no', async conv => {
   conv.contexts.delete('sendmessage_data');
   conv.ask('Message not sent. Is there anything else I can do for you?');
-  conv.ask(new Suggestions('No, that\'s all', 'Yes'));
+  conv.ask(new Suggestions("No, that's all", 'Yes'));
   conv.contexts.set('anything_else', 2);
 });
 
 /**
-* call.user
-*/
-app.intent('call.user', async (conv, {target}) => {
+ * call.user
+ */
+app.intent('call.user', async (conv, { target }) => {
   const circuit = await getCircuit(conv);
   if (!circuit) {
     return;
   }
 
-  let users = await circuit.searchUsers(target);
+  let users = await getUsers(circuit, conv, target);
 
-  if (!users.length) {
+  if (users.length !== 1) {
     conv.contexts.set('calluser_getuser', 5);
-    conv.ask(`I cannot find any user called ${target}. What's the name?`);
-    return;
-  }
-
-  if (users.length === 1) {
-    // One result found. Ask user for confirmation.
-    const name = users.length && users[0].displayName;
-    conv.ask(`<speak>Ready to call ${name}?</speak>`, new Suggestions('Yes', `No`));
-    conv.contexts.set('calluser_data', 5, {
-      email: users[0].emailAddress,
-      name: name
-    });
     return;
   }
 
-  // Multiple matches. Show suggestions of the first few matches.
-  users = users.slice(0, Math.min(7, users.length));
-
-  const suggestions = users.map(u => u.displayName);
-  conv.contexts.set('calluser_getuser', 5);
-
-  conv.ask(`More than one user found with name ${target}. What's the full name?`, new Suggestions(suggestions));
-  conv.ask(new Suggestions(suggestions));
+  // One result found. Ask user for confirmation.
+  const name = users.length && users[0].displayName;
+  conv.ask(`<speak>Ready to call ${name}?</speak>`, new Suggestions('Yes', `No`));
+  conv.contexts.set('calluser_data', 5, {
+    email: users[0].emailAddress,
+    name: name
+  });
 });
 
 /**
  * call.user - collect target
  */
-app.intent('call.user - collect target', async (conv, {target}) => {
+app.intent('call.user - collect target', async (conv, { target }) => {
   const circuit = await getCircuit(conv);
   if (!circuit) {
     return;
   }
 
-  let users = await circuit.searchUsers(target);
+  let users = await getUsers(circuit, conv, target);
```

Comment:
You are really searching for a user, not "getting" it. I'd prefer to call the local function "searchUsers"

Reply:
Done.